pkg/storage/fs/cback/utilities.go

```diff
@@ -0,0 +1,360 @@
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []snapshotResponse{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	return responseObject, nil
+}
+
+func (fs *cback) matchBackups(userName, pathInput string) (*backUpResponse, error) {
+
+	url := fs.conf.APIURL + "/backups/"
+	requestType := "GET"
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []backUpResponse{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	if len(responseObject) == 0 {
+		err = errors.New("no match found")
+		return nil, err
+	}
+
+	for i := range responseObject {
+		if responseObject[i].Detail != "" {
+			err = errors.New(responseObject[i].Detail)
+			return nil, err
+		}
+
+		if strings.Compare(pathInput, responseObject[i].Source) == 0 {
+			return &responseObject[i], nil
+		}
+	}
+
+	for i := range responseObject {
+		if responseObject[i].Detail != "" {
+			err = errors.New(responseObject[i].Detail)
+			return nil, err
+		}
+
+		if strings.HasPrefix(pathInput, responseObject[i].Source) {
+			substr := strings.TrimPrefix(pathInput, responseObject[i].Source)
+			substr = strings.TrimLeft(substr, "/")
+			responseObject[i].Substring = substr
+			return &responseObject[i], nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (fs *cback) statResource(backupID int, snapID, userName, path, source string) (*fsReturn, error) {
+
+	url := fs.conf.APIURL + "/backups/" + strconv.Itoa(backupID) + "/snapshots/" + snapID + "/" + path + "?content=false"
+	requestType := "OPTIONS"
+
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	responseObject := contents{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	m, err := mapReturn(responseObject.Type)
+
+	if err != nil {
+		return nil, err
+	}
+
+	retObject := fsReturn{
+		Path:   source + "/" + snapID + strings.TrimPrefix(responseObject.Name, source),
+		Type:   m,
+		Mtime:  uint64(responseObject.Mtime),
+		Size:   responseObject.Size,
+		Detail: responseObject.Detail,
+	}
+
+	return &retObject, nil
+}
+
+func (fs *cback) fileSystem(backupID int, snapID, userName, path, source string) ([]fsReturn, error) {
+
+	url := fs.conf.APIURL + "/backups/" + strconv.Itoa(backupID) + "/snapshots/" + snapID + "/" + path + "?content=true"
+	requestType := "OPTIONS"
+
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []contents{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	resp := make([]fsReturn, len(responseObject))
```

Comment:
change this to make([]*fsReturn, 0, len(responseObject))
Note the pointer

Reply:
Changed accordingly, and changed return type of function fileSystem to ([]*fsReturn, error)